src/main/recorder.ts

```diff
@@ -192,58 +244,52 @@ type RecorderOptionsType = {
             const isRaid = metadata.category == VideoCategory.Raids;
             const isLongEnough = (metadata.duration - overrun) >= this._options.minEncounterDuration;
 
-            if ((!isRaid || isLongEnough) && !discardVideo) {
-                // Cut the video to length and write its metadata JSON file.
-                // Await for this to finish before we return to waiting state.
-                await this.finalizeVideo(metadata, outputFilename);
-            } else {
+            if (isRaid && !isLongEnough) {
                 console.info("[Recorder] Raid encounter was too short, discarding");
+            } else if (discardVideo) {
+                console.info("[Recorder] Discarding video");
+            } else {
+                const bufferFile = obsRecorder.getObsLastRecording();
+                if (bufferFile) {
+                    await this.queueVideo(bufferFile, metadata);
+                } else {
+                    console.error("[Recorder] Unable to get the last recording from OBS. Can't process video.");
+                }
             }
 
-            // Run the size monitor to ensure we stay within size limit.
-            // Need some maths to convert GB to bytes
-            runSizeMonitor(this._options.storageDir, this._options.maxStorage)
-                .then(() => {
-                    if (mainWindow) mainWindow.webContents.send('refreshState');
-                });
-
-            // Clean-up the temporary recording directory. 
-            this.cleanupBuffer(1);
-
             // Refresh the GUI
             if (mainWindow) mainWindow.webContents.send('refreshState');
 
             // Restart the buffer recording ready for next game.
-            setTimeout(async () => {
-                this.startBuffer();
-            }, 5000)
+            this.startBuffer();
```

Comment:
How will the status icon behave here? Perhaps we want a second status icon for saving, so we can also show we've gone back to a watch state?

Reply:
As it did before. I've retained the status updates as `queueVideo()` _does_ have a slight delay of a few seconds which will be marked as "_Saving_".

Reply:
OK, but the saving icon will disappear before the video becomes available now. That's a bit unintuitive I think - I like the idea of having it as a second icon still.

Reply:
Yes, it will. How would you like it to display the status?

Reply:
That's a good question, and probably one I can work out myself with some frontend tinkering. I'm happy to merge this and handle that myself.

Reply:
I was thinking to expand the status thing at some point anyway. I'd like it to show some information about what it's recording and such, like perhaps "_Tazavesh: Streets +17_" or the name of a raid boss - perhaps we could show an icon somewhere when a saving job starts and remove it when the queue becomes idle.

Reply:
Think there is potential for something really nice too. For now I'm probably going to make a quick fix to display the saving icon as a secondary but we should discuss a future enhancement!